osc-server/src/test/java/org/osc/core/broker/service/servicefunctionchain/ServiceFunctionChainRequestValidatorTest.java

```diff
@@ -0,0 +1,276 @@
+/*******************************************************************************
+ * Copyright (c) Intel Corporation
+ * Copyright (c) 2017
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.osc.core.broker.service.servicefunctionchain;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.lang.StringUtils;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.osc.core.broker.model.entities.virtualization.ServiceFunctionChain;
+import org.osc.core.broker.model.entities.virtualization.VirtualizationConnector;
+import org.osc.core.broker.service.dto.BaseDto;
+import org.osc.core.broker.service.exceptions.VmidcBrokerInvalidEntryException;
+import org.osc.core.broker.service.exceptions.VmidcBrokerInvalidRequestException;
+import org.osc.core.broker.service.exceptions.VmidcBrokerValidationException;
+import org.osc.core.broker.service.request.AddOrUpdateServiceFunctionChainRequest;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ServiceFunctionChainRequestValidatorTest extends BaseServiceFunctionChainRequestValidatorTest {
+
+	@Test
+	public void testValidate_ThrowsNullPointerException() throws Exception {
+
+		// Arrange.
+		this.exception.expect(NullPointerException.class);
+
+		// Act.
+		this.validator.validate(null);
+	}
+
+	@Test
+	public void testValidate_WithNullNameField_ThrowsVmidcBrokerInvalidEntryException() throws Exception {
+
+		AddOrUpdateServiceFunctionChainRequest request = new AddOrUpdateServiceFunctionChainRequest();
+		BaseDto dto = new BaseDto();
+		request.setDto(dto);
+		// Arrange.
+		this.exception.expect(VmidcBrokerInvalidEntryException.class);
+
+		// Act.
+		this.validator.validate(request);
+	}
+
+	@Test
+	public void testValidate_WithInvalidFieldLenght_ThrowsVmidcBrokerInvalidEntryException() throws Exception {
```

Comment:
whats the difference between this method and next one testValidate_WithInvalidNameLenghtMax_ThrowsVmidcBrokerValidationException()?

Reply:
When Name field is validated it check for max-length i.e 155 and other test case is if maximum string length cannot be more than 15.  Hence added two test cases